lib/internal/child_process.js

```diff
@@ -516,6 +529,15 @@ ChildProcess.prototype.kill = function(sig) {
   return false;
 };
 
+ChildProcess.prototype[SymbolAsyncDispose] = async function() {
+  if (!this.closed) {
+    const ret = this.kill(this[kKillSignal]);
+    assert(ret, 'unexpected kill failure');
+    await EventEmitter.once(this, 'close');
+    this.emit('error', new AbortError());
```

Comment:
Why this? Not saying I disagree. Just not sure whether I do agree. @benjamingr any thoughts?

Reply:
So other consumers of the child process know it was aborted. this is the same behavior we've implemented on `Readable[Symbol.asyncDispose]`

Reply:
There is a race here though. The process might still exit successfully.

Reply:
I am Not sure I follow. can you provide an example?

Reply:
The process can succeed before it receives the kill. In which case you are emitting a bad error.

Reply:
An abortion is just a hint. The process may still choose the fully complete.

Reply:
I think this quickly becomes complicated. I'd prefer not to emit any error here.

Reply:
it can be a surprising behavior for a child process to exit without a kill signal and a code, and with incomplete stdout/stderr

Reply:
Let's ask for @ronbuckton 's advice:

The question here is:

```js
await using cp = childProcess1;

// someOtherCode

doSomethingWith(childProcess1);
```

What should the behavior be from doSomethingWith's side? For example if someone is reading stdout while another consumer is disposing it should it error?